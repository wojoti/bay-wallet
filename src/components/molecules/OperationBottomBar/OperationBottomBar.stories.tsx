import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {OperationBottomBar} from 'components/molecules';

export default {
  title: 'Molecules/OperationBottomBar',
  component: OperationBottomBar,
  argTypes: {
    onCardPress: {action: 'onCardPress'},
    onMenuPress: {action: 'onMenuPress'},
    onChartPress: {action: 'onChartPress'},
  },
} as ComponentMeta<typeof OperationBottomBar>;
const Template: ComponentStory<typeof OperationBottomBar> = args => (
  <OperationBottomBar {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
