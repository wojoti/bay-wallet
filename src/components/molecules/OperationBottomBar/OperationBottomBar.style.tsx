import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    width: '100%',
    height: 200,
    right: 0,
    bottom: 20,
    left: 0,
  },
});
export default style;
