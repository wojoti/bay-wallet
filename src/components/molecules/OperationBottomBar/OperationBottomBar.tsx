import {ComposableButton, Container, Img, ImgType} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './OperationBottomBar.style';

export type OperationBottomBarProps = PropsWithChildren<{
  onCardPress: () => void;
  onMenuPress: () => void;
  onChartPress: () => void;
  testId?: string;
}>;

const OperationBottomBar = ({
  onCardPress,
  onMenuPress,
  onChartPress,
  testId,
}: OperationBottomBarProps) => {
  return (
    <Container testId={testId} customStyle={style.wrapper} align="center">
      <Container
        direction="row"
        justify="space-between"
        align="center"
        customStyle={style.whiteArea}>
        <ComposableButton
          customStyle={style.button}
          onPress={onCardPress}
          testId={'operationbottombar-cards'}>
          <Img src={ImgType.iconCard} height={'17'} width={23} alt={''} />
        </ComposableButton>
        <ComposableButton
          customStyle={style.purpleButton}
          onPress={onMenuPress}
          testId={'operationbottombar-menu'}>
          <Img src={ImgType.squares} height={'30'} width={30} alt={''} />
        </ComposableButton>
        <ComposableButton
          customStyle={style.button}
          onPress={onChartPress}
          testId={'operationbottombar-chart'}>
          <Img src={ImgType.iconChart} height={'31'} width={23} alt={''} />
        </ComposableButton>
      </Container>
    </Container>
  );
};

export default OperationBottomBar;
