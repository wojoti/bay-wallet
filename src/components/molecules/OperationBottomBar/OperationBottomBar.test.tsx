import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import OperationBottomBar, {
  OperationBottomBarProps,
} from './OperationBottomBar';

const mockOnCardPress = jest.fn();
const mockOnMenuPress = jest.fn();
const mockOnChartPress = jest.fn();
const props: OperationBottomBarProps = {
  testId: 'test-operationbottombar-id',
  onCardPress: mockOnCardPress,
  onMenuPress: mockOnMenuPress,
  onChartPress: mockOnChartPress,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<OperationBottomBar {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render OperationBottomBar', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
});

test('should render OperationBottomBar - cards button', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-cards',
  );
  expect(buttonElement).toBeOnTheScreen();
});
test('should render OperationBottomBar - menu button', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-menu',
  );
  expect(buttonElement).toBeOnTheScreen();
});
test('should render OperationBottomBar - chart button', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-chart',
  );
  expect(buttonElement).toBeOnTheScreen();
});

test('should render OperationBottomBar - cards button - handle onPress', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-cards',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(mockOnCardPress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(mockOnCardPress).toBeCalledTimes(1);
});
test('should render OperationBottomBar - menu button - handle onPress', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-menu',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(mockOnMenuPress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(mockOnMenuPress).toBeCalledTimes(1);
});
test('should render OperationBottomBar - chart button - handle onPress', () => {
  render(<OperationBottomBar {...props} />);
  const OperationBottomBarElement = screen.getByTestId(
    'test-operationbottombar-id',
  );
  expect(OperationBottomBarElement).toBeOnTheScreen();
  const buttonElement = within(OperationBottomBarElement).getByTestId(
    'operationbottombar-chart',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(mockOnChartPress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(mockOnChartPress).toBeCalledTimes(1);
});
