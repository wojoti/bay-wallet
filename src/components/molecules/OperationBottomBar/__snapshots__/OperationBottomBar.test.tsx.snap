// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "#ffffffab",
        "bottom": 0,
        "left": 0,
        "paddingBottom": 45,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      },
    ]
  }
  testID="test-operationbottombar-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": undefined,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        },
        {
          "backgroundColor": "white",
          "borderRadius": 16,
          "elevation": 10,
          "width": "85%",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "paddingHorizontal": 40,
            "paddingVertical": 20,
          },
        ]
      }
      testID="operationbottombar-cards"
    >
      <SvgMock
        height="17"
        width={23}
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "backgroundColor": "#363062",
            "borderRadius": 45,
            "bottom": 30,
            "elevation": 10,
            "padding": 20,
          },
        ]
      }
      testID="operationbottombar-menu"
    >
      <SvgMock
        height="30"
        width={30}
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "paddingHorizontal": 40,
            "paddingVertical": 20,
          },
        ]
      }
      testID="operationbottombar-chart"
    >
      <SvgMock
        height="31"
        width={23}
      />
    </View>
  </View>
</View>
`;
